index.js

```diff
@@ -841,6 +842,37 @@ RingPop.prototype.handleOrProxy =
         }
     };
 
+RingPop.prototype.handleOrProxyAll =
+    function handleOrProxyAll(opts) {
+        var self = this;
+        var keys = opts.keys;
+        var req = opts.req;
+        var localHandler = opts.localHandler;
+
+        var reses = {};
+        var dests = mapUniq(keys, this.lookup.bind(this));
+        dests.forEach(function(dest) {
+            var res = hammock.Response();
+            if (self.whoami() === dest) {
+                process.nextTick(localHandler.bind(null, req, res));
+            } else {
+                process.nextTick(self.proxyReq.bind(self, _.defaults({
+                    dest: dest,
+                    res: res
+                }, opts)));
```

Comment:
is there a reason not to be explicit with the opts that we're sending to proxyReq and just providing all of `opts`? This'll pass over `localHandler` which is obviously unnecessary.